internal/k8shandler/forwarding.go

```diff
@@ -39,59 +40,69 @@ func EvaluateAnnotationsForEnabledCapabilities(forwarder *logging.ClusterLogForw
-		return "", nil
-	}
-	switch clusterRequest.Cluster.Spec.Collection.Type {
-	case logging.LogCollectionTypeFluentd:
-		break
-	case logging.LogCollectionTypeVector:
-		break
-	default:
-		return "", fmt.Errorf("%s collector does not support pipelines feature", clusterRequest.Cluster.Spec.Collection.Type)
-	}
 
-	if clusterRequest.Forwarder == nil {
-		clusterRequest.Forwarder = &logging.ClusterLogForwarder{}
+	var collectionSpec *logging.CollectionSpec
+	// If clusterLogging is specified, check if collector type is valid
+	if clusterRequest.Cluster != nil && clusterRequest.Cluster.Name == constants.SingletonName {
+		if clusterRequest.Cluster.Spec.Collection == nil {
+			log.V(2).Info("skipping collection config generation as 'collection' section is not specified in CLO's CR")
+			return "", nil
+		}
+		switch clusterRequest.Cluster.Spec.Collection.Type {
+		case logging.LogCollectionTypeFluentd:
+			break
+		case logging.LogCollectionTypeVector:
+			break
+		default:
+			return "", fmt.Errorf("%s collector does not support pipelines feature", clusterRequest.Cluster.Spec.Collection.Type)
+		}
+		collectionSpec = clusterRequest.Cluster.Spec.Collection
+	} else {
+		// Make a default collection spec with just the type
+		collectionSpec = &logging.CollectionSpec{
+			Type: logging.LogCollectionTypeVector,
+		}
 	}
 
 	// Set the output secrets if any
 	clusterRequest.SetOutputSecrets()
-
 	tokenSecret, err := clusterRequest.getLogCollectorServiceAccountTokenSecret()
 	if err == nil {
-		clusterRequest.OutputSecrets[constants.LogCollectorToken] = tokenSecret
+		saTokenSecretName := clusterRequest.ResourceNames[constants.SaTokenSecret]
+		clusterRequest.OutputSecrets[saTokenSecretName] = tokenSecret
 	}
 
 	op := generator.Options{}
 	tlsProfile, _ := tls.FetchAPIServerTlsProfile(clusterRequest.Client)
 	op[generator.ClusterTLSProfileSpec] = tls.GetClusterTLSProfileSpec(tlsProfile)
 	EvaluateAnnotationsForEnabledCapabilities(clusterRequest.Forwarder, op)
 
-	var collectorType = clusterRequest.Cluster.Spec.Collection.Type
-	g := forwardergenerator.New(collectorType)
-	generatedConfig, err := g.GenerateConf(clusterRequest.Cluster.Spec.Collection, clusterRequest.OutputSecrets, &clusterRequest.Forwarder.Spec, clusterRequest.Cluster.Namespace, op)
+	g := forwardergenerator.New(collectionSpec.Type)
+	generatedConfig, err := g.GenerateConf(collectionSpec, clusterRequest.OutputSecrets, &clusterRequest.Forwarder.Spec, constants.WatchNamespace, op)
 
 	if err != nil {
 		log.Error(err, "Unable to generate log configuration")
-		if updateError := clusterRequest.UpdateCondition(
-			logging.CollectorDeadEnd,
-			"Collectors are defined but there is no defined LogStore or LogForward destinations",
-			"No defined logstore destination",
-			corev1.ConditionTrue,
-		); updateError != nil {
-			log.Error(updateError, "Unable to update the clusterlogging status", "conditionType", logging.CollectorDeadEnd)
+		if clusterRequest.Cluster != nil {
```

Comment:
Does this already exist in our validators.  There is no reason it should not or can not

Reply:
I'm inclined to remove this status condition as it doesn't accurately reflect what is going on. This status gets updated upon generating the fluentd or vector config and should reflect configuration failures instead.